services/users/src/routes/team.ts

```diff
@@ -0,0 +1,102 @@
+import express from "express";
+import { asyncHandler } from "@api/common/src/middleware";
+import { BadRequestError } from "@api/common/src/errors";
+
+import { TeamModel, Team } from "../models/team";
+
+export const teamRoutes = express.Router();
+
+teamRoutes.route("/").post(
+  asyncHandler(async (req, res) => {
+    const { name, event, members, description, publicTeam } = req.body;
+
+    const team = await TeamModel.findOne({ name, event });
+
+    if (team) {
+      throw new BadRequestError("Team with this name already exists!");
+    }
+
+    const teams = await TeamModel.find({ event, members: req.user?.uid });
```

Comment:
Does this filter work like this? dont u have to specify with mongoose commands like if the value exists in the array?

Reply:
<img width="814" alt="Screen Shot 2022-03-29 at 6 14 32 PM" src="https://user-images.githubusercontent.com/25470781/160715876-72a67051-3c9f-46bc-ba3f-0f1d649b06e3.png">

I think this would work for checking whether the array contains the element so it works as intended.

Reply:
oh ok hype thats cool didnt know that